source/hardware/controllers/behringer_bcr2000.rst

```diff
@@ -0,0 +1,106 @@
+following options:
+
+#. Load the dump file ``bcr_Only_Controllers.syx`` via Sysex to any other preset slot.
+   The dump file and instructions how to install it in the controller is available on the
+   manufacturer's webpage.
+#. Factory reset the controller.
+#. Customize the preset mapping (see section *Mapping Description* for details).
+
+Button Behaviour
+^^^^^^^^^^^^^^^^
+It is recommended, but not required, to change the button behaviour from *Toggle On* (default)
+to *Toggle Off*. Both behaviours send an ``On`` message on button press, but the action to
+send an ``Off`` message is different: *Toggle Off* responds to a button release while *Toggle On*
+ignores the release and requires a second press. To change the behaviour:
+
+#. Hold *Edit*, press a button, then release both
+#. Turn the 6th push encoder (labelled *MODE*) until the display shows ``toFF``
+#. Press *Exit*
+
+To make the change persistent, press *Store* twice. See the manual for details.
+
+.. _mapping-description:
+
+Mapping Description
+-------------------
+The BCR2000 is a general purpose controller that allows different mappings.
+
+The file ``Behringer-BCR2000-scripts.js`` contains a default mapping which works
+out-of-the-box for a factory-reset controller, but may be customized freely according to your needs.
+
+The mapping is split into several files:
+
+============================================== =================== ===========
+File                                           When to edit?       Description
+============================================== =================== ===========
+``Behringer-BCR2000-scripts.js``               Mapping changes     Mapping configuration. You can assign Mixxx controls to hardware controls here, and change which controls are bound to the Shift layer.
+``Behringer-BCR2000-preset-scripts.js``        Preset changes      Contains the mapping of MIDI messages to hardware controls. If for some reason you'd like to use a different preset than the factory settings of preset #1, you can change the MIDI addresses according to your controller's preset within this file.
+``Behinger BCR2000.midi.xml``                  Preset changes      Entry point that contains the MIDI addresses for all input controls and includes all other files. The mapping of MIDI messages to Mixxx controls is purely done in Javascript and based on `Components JS <https://github.com/mixxxdj/mixxx/wiki/Components%20JS>`_, so that the XML file does not contain any information about the Mixxx controls (e.g. `key` and `group`).
+``midi-components-extension.js``               Usually never       Contains required Javascript components for the mapping.
+============================================== =================== ===========
+
+Default Mapping
+^^^^^^^^^^^^^^^
+The default mapping covers 2 effect units and loop, reverse & pitch controls for 2 decks:
+
+.. figure:: behringer_bcr2000.svg
+  :width: 600
+
```

Comment:
Please add a table that explains how the mapping works. I suggest to explain that the left half of the two top rows is used for deck 1, and the right half for deck 2, then make a table that explains the knobs and buttons from left to right. If there's a label on the hardware, please mention it:

```rst
========================================  =========================================================================
Control                                   Description
========================================  =========================================================================
Knob 1 (:hwlabel:`TYPE`/:hwlabel:`VAL2`)  Manipulates the key of the track in semitones.
Knob 2 (:hwlabel:`CH`/:hwlabel:`MODE`)    Moves the current loop left or right.
...                                       ...
Button 1                                  Toggles Keylock.
Shift + Button 1                          Toggles Vinyl mode.
...                                       ...
========================================  =========================================================================
```

Reply:
I added a table, please check if it's well-arranged.